simpeg_drivers/electromagnetics/time_domain/driver.py

```diff
@@ -30,3 +36,64 @@ class TimeDomainElectromagneticsDriver(InversionDriver):
 
     def __init__(self, params: TimeDomainElectromagneticsParams):
         super().__init__(params)
+
+    def get_tiles(self):
+        """
+        Special method to tile the data based on the transmitters center locations.
+
+        First the transmitter locations are grouped into groups using kmeans clustering.
+        Second, if the number of groups is less than the number of 'tile_spatial' value, the groups are
+        further divided into groups based on the clustering of receiver locations.
+        """
+        if isinstance(self.params.data_object, LargeLoopGroundTEMReceivers):
+
```

Comment:
This method is pretty long.  I would at least break out the tem section to it's own method and call here.  Better still would be a tile_tem_transmitters, refine_tem_tiles(receiver_ids), that feed a tile_tem_survey() that get called at line 49 here.

Reply:
I would also take a look at simplifying the naming of all the index arrays since the split might let you keep natural naming (eg: t_id -> tx_id).

Reply:
..and uids is maybe a poor choice.  unique_tx_ids?

Reply:
Made simplifications. See what you think

Reply:
yeah, that's better.  I still think there's a couple/few separate methods hidden in there, but I'll let it go for now.